sessionctx/variable/sysvar.go

```diff
@@ -1438,6 +1438,15 @@ var defaultSysVars = []*SysVar{
 		s.MemQuotaQuery = TidbOptInt64(val, DefTiDBMemQuotaQuery)
 		return nil
 	}},
+	{Scope: ScopeGlobal, Name: TiDBMemQuotaAnalyze, Value: strconv.Itoa(DefTiDBMemQuotaAnalyze), skipInit: true, Type: TypeInt, MinValue: 0, MaxValue: math.MaxInt32,
```

Comment:
```suggestion
	{Scope: ScopeGlobal, Name: TiDBMemQuotaAnalyze, Value: strconv.Itoa(DefTiDBMemQuotaAnalyze), Type: TypeInt, MinValue: 0, MaxValue: math.MaxInt32,
```

Reply:
Because the default is `-1`, but the MinValue is `0` you will also need to add: `AllowAutoValue: true`.

Reply:
Done